### PR TITLE
securitycontext: move docker-specific logic into kubelet/dockertools

### PR DESCRIPTION
**Pull request by yujuhong**
This change moves the code specific to docker to kubelet/dockertools,
while leaving the common utility functions at its current package
(pkg/securitycontext).

When we deprecate dockertools in the future, the code will be moved to
pkg/kubelet/dockershim instead.

### CODE REVIEW

**Comment by yujuhong** [comment]
There should be no functional change

@pweil- are you ok with the change?

File: pkg/kubelet/rkt/rkt.go [hunk 1]
```diff
@@ -2449,3 +2449,18 @@ func convertKubePortMappings(portMappings []kubecontainer.PortMapping) ([]appcty
 
 	return containerPorts, hostPorts
 }
+
+// makeCapabilities creates string slices from Capability slices
+func makeCapabilities(capAdd []v1.Capability, capDrop []v1.Capability) ([]string, []string) {
```
Comment (pweil-) [inline pkg/kubelet/rkt/rkt.go]: looks like `MakeCapabilities` is in the docker tools version as well still.  Should it be left in the common code and removed from this impl and the docker impl?

**Review by pweil- — APPROVED** [review]
One question but LGTM (I don't have label permissions)

File: pkg/kubelet/rkt/rkt.go (hunk 1 above)
Comment (yujuhong) [inline pkg/kubelet/rkt/rkt.go]: I moved it to `pkg/kubelet/container` to be shared by both dockertools and rkt. This doesn't seem like a function that non-kubelet packages will be interested in, and we will be deprecating dockertools and rkt in the future. Are you ok with that?

Comment (pweil-) [inline pkg/kubelet/rkt/rkt.go]: SGTM.  Thanks!

Comment (yujuhong) [inline pkg/kubelet/rkt/rkt.go]: Squashed. Thanks!

**Comment by yujuhong** [comment]
Applying the lgtm label based on https://github.com/kubernetes/kubernetes/pull/40574#pullrequestreview-18836784

**Comment by yujuhong** [comment]
@pmorie I'll help with your rebase if needed, but let's get this one in to avoid more unnecessary conflict.

**Comment by yujuhong** [comment]
@dchen1107 @thockin, can one of you help approve this? Thanks!

**Comment by dchen1107** [comment]
/lgtm
and 
/approve

**Pull request merged**